src/Receipts/Receipt.php

```diff
@@ -0,0 +1,306 @@
+     *
+     * @return $this
+     */
+    public function setIsBso(?bool $is_bso): self
+    {
+        $this->is_bso = $is_bso;
+
+        return $this;
+    }
+
+    /**
+     * @return float|null
+     */
+    public function getElectronicAmount(): ?float
+    {
+        return $this->electronic_amount;
+    }
+
+    /**
+     * @param float|null $electronic_amount
+     *
+     * @return $this
+     */
+    public function setElectronicAmount(?float $electronic_amount): self
+    {
+        $this->electronic_amount = $electronic_amount;
+
+        return $this;
+    }
+
+    /**
+     * @return float|null
+     */
+    public function getAdvancePaymentAmount(): ?float
+    {
+        return $this->advance_payment_amount;
+    }
+
+    /**
+     * @param float|null $advance_payment_amount
+     *
+     * @return $this
+     */
+    public function setAdvancePaymentAmount(?float $advance_payment_amount): self
+    {
+        $this->advance_payment_amount = $advance_payment_amount;
+
+        return $this;
+    }
+
+    /**
+     * @return float|null
+     */
+    public function getCreditAmount(): ?float
+    {
+        return $this->credit_amount;
+    }
+
+    /**
+     * @param float|null $credit_amount
+     *
+     * @return $this
+     */
+    public function setCreditAmount(?float $credit_amount): self
+    {
+        $this->credit_amount = $credit_amount;
+
+        return $this;
+    }
+
+    /**
+     * @return float|null
+     */
+    public function getProvisionAmount(): ?float
+    {
+        return $this->provision_amount;
+    }
+
+    /**
+     * @param float|null $provision_amount
+     *
+     * @return $this
+     */
+    public function setProvisionAmount(?float $provision_amount): self
+    {
+        $this->provision_amount = $provision_amount;
+
+        return $this;
+    }
+
+    /**
+     * @return array
+     */
+    public function toArray(): array
+    {
+        $receipt_items = [];
+        foreach ($this->items as $receipt_item) {
+            if ($receipt_item instanceof Item) {
+                $receipt_items[] = $receipt_item->toArray();
+            }
+        }
+
+        return \array_filter([
+            'Items'            => $receipt_items,
+            'calculationPlace' => $this->calculation_place,
+            'taxationSystem'   => $this->taxation_system,
+            'email'            => $this->email,
+            'phone'            => $this->phone,
+            'isBso'            => $this->is_bso,
+            'amounts'          => [
+                'electronic'     => \number_format((float) $this->electronic_amount, 2, '.', ''),
+                'advancePayment' => \number_format((float) $this->advance_payment_amount, 2, '.', ''),
+                'credit'         => \number_format((float) $this->credit_amount, 2, '.', ''),
+                'provision'      => \number_format((float) $this->provision_amount, 2, '.', ''),
+            ],
+        ], static function ($value) {
```

Comment:
as idea: `\\is_null`

Reply:
nope, we need filter **not** null values